### DISCUSSION TITLE
md-select-option selected not getting selected

### DISCUSSION THREAD

**Discussion by srdavo** [material-components/material-web#5584] [Q&A]
### What is affected?

Component

### Description

I have some "md-outlined-select"or "md-filled-select" in my project and inside there are md-select-option with the 
attribute, sometimes when the page loads the select does not have the option selected and some other times it does show the option selected.
Some times it does not show the option selected but the select does have the value and some other times the select does not show the option selected and also does not have a value, so it makes it unusable.

I tried with material web dialogs, with my own modals, and the problem persists.
https://github.com/material-components/material-web/assets/97427028/6d125544-48c4-4721-9e38-d0d64712589f

### Reproduction

Sadly i cant reproduce it, so i guess it is something with me, but if any one could help with this weird problem. 
https://lit.dev/playground/#project=W3sibmFtZSI6Im1hdGVyaWFsLWltcG9ydHMuanMiLCJjb250ZW50IjoiaW1wb3J0ICdAbWF0ZXJpYWwvd2ViL3NlbGVjdC9maWxsZWQtc2VsZWN0LmpzJztcbmltcG9ydCAnQG1hdGVyaWFsL3dlYi9zZWxlY3Qvb3V0bGluZWQtc2VsZWN0LmpzJztcbmltcG9ydCAnQG1hdGVyaWFsL3dlYi9zZWxlY3Qvc2VsZWN0LW9wdGlvbi5qcyc7In0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vbWF0ZXJpYWwtaW1wb3J0cy5qc1wiPjwvc2NyaXB0PlxuXG5cbjxtZC1vdXRsaW5lZC1zZWxlY3RcbiAgICAgICAgaWQ9XCJmaWx0ZXItYXBwdHMteWVhclwiXG4gICAgICAgIGxhYmVsPVwiQcOxbyBkZSBsYXMgY2l0YXNcIlxuICAgICAgICBjbGFzcz1cIm5vLXJlc2V0XCI-XG5cbiAgICAgICAgPG1kLXNlbGVjdC1vcHRpb24gc2VsZWN0ZWQgdmFsdWU9XCIyMDI0XCI-XG4gICAgICAgICAgMjAyNFxuICAgICAgICA8L21kLXNlbGVjdC1vcHRpb24-XG4gICAgICAgIDxtZC1zZWxlY3Qtb3B0aW9uIHZhbHVlPVwiMjAyM1wiPlxuICAgICAgICAgIDIwMjNcbiAgICAgICAgPC9tZC1zZWxlY3Qtb3B0aW9uPlxuICAgICAgICA8bWQtc2VsZWN0LW9wdGlvbiB2YWx1ZT1cIjIwMjIsIDIwMjMsIDIwMjRcIj5cbiAgICAgICAgICBUb2RvcyBsb3MgYcOxb3NcbiAgICAgICAgPC9tZC1zZWxlY3Qtb3B0aW9uPlxuICAgICAgPC9tZC1vdXRsaW5lZC1zZWxlY3Q-In0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMS4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4yLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9XQ

### Workaround

i will just change the way my app works to use this.

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

1.4.1

### Browser/OS/Node environment

Chrome, last version

**Comment by asyncliz** [comment]
Moving to a discussion so folks can help investigate and answer
